### PR TITLE
Migrate from slog to tracing.

### PR DESCRIPTION
**Pull request by mental32**

### CODE REVIEW

File: database/src/lib.rs [hunk 1]
```diff
@@ -170,10 +168,7 @@ async fn internal_get_conn_custom(
     let pool = sqlx::Pool::connect("postgres://postgres:dimpostgres@127.0.0.1/").await;
 
     if let Some(log) = log {
```
Comment (vgarleanu) [inline database/src/lib.rs:170]: This if let wont be needed anymore.

File: database/src/lib.rs [hunk 2]
```diff
@@ -157,10 +158,7 @@ async fn internal_get_conn(_log: Option<&Logger>) -> sqlx::Result<DbConnection>
 
 #[cfg(feature = "postgres")]
 #[async_recursion::async_recursion]
-async fn internal_get_conn_custom(
-    log: Option<&'async_recursion Logger>,
-    main: &str,
-) -> sqlx::Result<DbConnection> {
+async fn internal_get_conn_custom(main: &str) -> sqlx::Result<DbConnection> {
```
Comment (vgarleanu) [inline database/src/lib.rs:161]: This function should also derive instrument i think

File: dim/src/scanners/tv_show.rs [hunk 3]
```diff
@@ -75,11 +72,11 @@ impl<'a> TvShowMatcher<'a> {
                     Ok(x) => Some(x.id),
                     Err(e) => {
                         warn!(
-                            self.log,
-                            "Failed to insert poster into db";
-                            "reason" => e.to_string(),
-                            "orphan_id" => orphan.id
+                            "Failed to insert poster into db {}/{}",
+                            reason = e.to_string(),
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:76]: instead of calling `to_string`, we can use the `?` operator

File: dim/src/scanners/tv_show.rs [hunk 4]
```diff
@@ -105,10 +102,9 @@ impl<'a> TvShowMatcher<'a> {
                     Ok(x) => Some(x.id),
                     Err(e) => {
                         warn!(
-                            self.log,
-                            "Failed to insert backdrop into db";
-                            "reason" => e.to_string(),
-                            "orphan_id" => orphan.id
+                            "Failed to insert backdrop into db {}/{}",
+                            reason = e.to_string(),
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:106]: same as above

File: dim/src/scanners/tv_show.rs [hunk 5]
```diff
@@ -131,10 +127,9 @@ impl<'a> TvShowMatcher<'a> {
 
         if let Err(e) = self.insert(orphan, media, result).await {
             warn!(
-                self.log,
-                "Failed to insert new media";
-                "id" => orphan.id,
-                "reason" => e.to_string(),
+                "Failed to insert new media {}/{}",
+                id = orphan.id,
+                reason = e.to_string(),
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:132]: same as above

File: dim/src/scanners/tv_show.rs [hunk 6]
```diff
@@ -190,10 +185,9 @@ impl<'a> TvShowMatcher<'a> {
                     Ok(x) => Some(x.id),
                     Err(e) => {
                         warn!(
-                            self.log,
-                            "Failed to insert season poster into db";
-                            "reason" => e.to_string(),
-                            "orphan_id" => orphan.id
+                            "Failed to insert season poster into db {}/{}",
+                            reason = e.to_string(),
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:189]: same as above

File: dim/src/scanners/tv_show.rs [hunk 7]
```diff
@@ -211,7 +205,10 @@ impl<'a> TvShowMatcher<'a> {
         let seasonid = match insertable_season.insert(&self.conn, media_id).await {
             Ok(x) => x,
             Err(e) => {
-                warn!(self.log, "Failed to insert season into the database."; "reason" => e.to_string());
+                warn!(
+                    "Failed to insert season into the database. {}",
+                    reason = e.to_string()
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:210]: same as above

File: dim/src/scanners/tv_show.rs [hunk 8]
```diff
@@ -249,10 +246,9 @@ impl<'a> TvShowMatcher<'a> {
                     Ok(x) => Some(x.id),
                     Err(e) => {
                         warn!(
-                            self.log,
-                            "Failed to insert still into db";
-                            "reason" => e.to_string(),
-                            "orphan_id" => orphan.id
+                            "Failed to insert still into db {}/{}",
+                            reason = e.to_string(),
```
Comment (vgarleanu) [inline dim/src/scanners/tv_show.rs:250]: same as above

**Comment by mental32** [comment]
@vgarleanu updated

**Comment by vgarleanu** [comment]
Dim doesnt log anything to stdout anymore.

**Comment by songb1rd** [comment]
Set the RUST_LOG environment variable

File: dim/src/core.rs [hunk 9]
```diff
@@ -32,25 +30,25 @@ pub static METADATA_PATH: OnceCell<String> = OnceCell::new();
 /// * `log` - Logger to which to log shit
 /// * `tx` - this is the websocket channel to which we can send websocket events to which get
 /// dispatched to clients.
-pub async fn run_scanners(log: Logger, tx: EventTx) {
-    if let Ok(conn) = database::get_conn_logged(&log).await {
+#[instrument]
```
Comment (vgarleanu) [inline dim/src/core.rs:33]:
```suggestion
#[instrument(skip_all)]
```

File: dim/src/core.rs [hunk 10]
```diff
@@ -61,8 +59,8 @@ pub async fn run_scanners(log: Logger, tx: EventTx) {
     }
 }
 
+#[instrument(skip(stream_manager))]
```
Comment (vgarleanu) [inline dim/src/core.rs:62]:
```suggestion
#[instrument(skip(stream_manager, event_tx, rt, event_rx))]
```

File: dim/src/logger.rs [hunk 11]
```diff
@@ -1,18 +1,13 @@
...
-}
+pub struct RequestLogger;
 
 impl RequestLogger {
-    pub fn new(logger: slog::Logger) -> Self {
-        Self {
-            logger: logger.new(o!("mod" => "warp")),
-        }
+    pub fn new() -> Self {
+        Self {}
```
Comment (vgarleanu) [inline dim/src/logger.rs:10]:
```suggestion
        Self
```

File: dim/src/scanners/mod.rs [hunk 12]
```diff
@@ -61,34 +61,38 @@ pub(super) static METADATA_EXTRACTOR: OnceCell<base::MetadataExtractor> = OnceCe
...
+        base::MetadataMatcher::cluster(&mut handle, 6, conn.clone(), tx.clone()).1
     })
 }
 
 pub fn get_matcher_unchecked() -> &'static base::MetadataMatcher {
     METADATA_MATCHER.get().unwrap()
 }
 
-pub async fn start_custom(
+#[instrument(skip(paths))]
```
Comment (vgarleanu) [inline dim/src/scanners/mod.rs:83]:
```suggestion
#[instrument(skip(tx, paths))]
```

**Pull request merged**